## Task 12 — Customer Activity by Country

### Business Question

Which countries have the highest average customer activity?

### Analysis

Group customers by country, calculate the number of customers, average download traffic, and average loyalty points for each country, then sort the results by average download traffic in descending order.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
customer_activity = starlink.groupby("country").agg(
    users_amount = ("unique_id","count"),
    gb_downloaded_average = ("gb_downloaded","mean"),
    loyalty_points_average = ("loyalty_points","mean")
).sort_values(
    "gb_downloaded_average",
    ascending = False
)
print(customer_activity)

                users_amount  gb_downloaded_average  loyalty_points_average
country                                                                    
United Kingdom           977            1573.757963                     NaN
Ukraine                 1032            1562.133547                   100.0
Germany                 1027            1534.529747                     NaN
France                   996            1526.645211                     NaN
Spain                   1019            1526.638440                     NaN
Australia               1030            1517.334117                     NaN
United States            954            1505.038040                     NaN
Canada                   997            1496.716038                     NaN
Brazil                   996            1465.334006                     NaN
Japan                    973            1450.620822                     NaN


### Result

The United Kingdom has the highest average download traffic at 1,573.76 GB per customer, followed by Ukraine at 1,562.13 GB and Germany at 1,534.53 GB. Loyalty points data is unavailable for most countries due to missing values.

### Business Insight

Average download traffic is relatively high across all analyzed countries, with the United Kingdom showing the highest value. The limited availability of loyalty points data prevents a meaningful comparison of customer loyalty across countries and indicates that this metric requires more complete data before further analysis.